### DISCUSSION TITLE
German spoken date is output with "dot" as text

### DISCUSSION THREAD

**Discussion by DZipfel1** [deepgram/community#797] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

For alle German number formats that contain dots (e.g. dates) deepgram transcript adds some "dot" which was never spoken. German example of spoken text: "Am siebenundzwanzigsten fünften vierundzwanzig"
Transcript: "am siebenundzwanzig punkt vier.2024"

This seems to recognize that this is a date with a dot as a separator. But instead of transcribing it as the date with numbers 27.4. (which is not yet available for German), it returns the first dot as text.

Additional info: problem only occurs if the date is said as numbers like "Dritter Elfter"  but not when saying an explicit month name like in "Dritter November".  Additional testing with "Dreiundzwanzigster Elfter" resultet in transcript " Dreinundzanzig Punkt eins eins", separating even the spoken "eleven" into "one one" as you can see in request ID: b666dc0c-59b7-4be3-8ae4-5105fb9f82e7

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?smart_format=true&model=nova-2&language=de&detect_language=false

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

dd7c4a57-8a1f-4cc8-aadd-9e145942df05

### If possible, please attach your code or paste it into the text box.

```
async function transcribeAudio(URL) {
    try{
        const transcriptionFeatures = { smart_format: true, model: 'nova-2', language: 'de', detect_language: 'false' };
        const { results } = await deepgram.transcription.preRecorded({ url: URL }, transcriptionFeatures);
        return results.channels[0].alternatives[0].transcript;
    } catch (error) {
        console.error('Error transcribing audio:', error);
        throw error; // Rethrow or handle as needed
    }
}
```

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by SandraRodgers** [comment, marked as answer]
Thank you for reporting this. If I understand correctly, the problem is the following:

The Nova-2 model seems to be overly influenced by written date conventions in German, where dots are used to separate day, month, and year in German dates (e.g., 27.05.2024). When users speak dates in numerical format (e.g., "siebenundzwanzigster fünfter"), the model incorrectly inserts the word "Punkt" (dot) into the transcript, misunderstanding the spoken format (e.g. "am siebenundzwanzig punkt vier.2024")

To me this seems to be a problem with the model's training. It needs better training on German dates. I have reported this to our product/research/engineering teams.

For now there is no solution but we will work to improve the model. Thank you for letting us know!

**Comment by DZipfel1** [comment]
Yes, the model seems to do the following:

1. speech to text "siebenundzwanzigster fünfte zwanzigvierundzwanzig" (my guess, since this state is not visible)
2. date recognition with conversion into German format like "27.05.2024" (my guess, since this state is not visible)
3. take the information from step 2 and reconvert into text like "Siebenundzwanzig Punkt fünf punkt zweitausendvierundzwanzig"
with some variations like

With this behavior Deepgram Nova-2 currently is not ready for production usage in Germany which i had just discussed with Anthony Minessale from Signalwire, who tries out Deepgram for their European solution.

**Comment by SandraRodgers** [comment]
I understand. We are always working towards improving our models. This is very useful information and will be used to help Deepgram improve. Thank you.
